File: javapatterns/src/main/java/org/iut/nantes/factories/cours/EDT.java
package org.iut.nantes.factories.cours;

class EDT
{
  private FactoryCours fc_;

  private Cours cours_;
  private Prof prof_;
  private Salle salle_;

  public EDT( FactoryCours fc )
  {
    fc_ = fc;
  }

  public void setFactory( FactoryCours fc )
  {
    fc_ = fc;
  }

  public void remplirEDT( String nom )
  {
    cours_ = fc_.creerCours();
    prof_ = fc_.creerProf( nom );
    salle_ = fc_.creerSalle();
  }

  public void afficherEDT()
  {
    if( prof_ != null )
      System.out.println( "M ou Mme " + prof_.getNom() + " du département " + prof_.getDomaine()
			  + " a la " + salle_.getType() + " numero " + salle_.getNumero()
			  + " pour le cours de " + cours_.getMatiere() );
  }
  
}

File: kotlinpatterns/src/main/kotlin/org/iut/nantes/factories/cours/EDT.kt
package org.iut.nantes.factories.cours

internal class EDT(private var fc_: FactoryCours) {
    private var cours_: Cours? = null
    private var prof_: Prof? = null
    private var salle_: Salle? = null
    fun setFactory(fc: FactoryCours) {
        fc_ = fc
    }

    fun remplirEDT(nom: String) {
        cours_ = fc_.creerCours()
        prof_ = fc_.creerProf(nom)
        salle_ = fc_.creerSalle()
    }

    fun afficherEDT() {
        println("M ou Mme ${prof_!!.nom} du département ${prof_!!.domaine}"
            + " a la ${salle_!!.type} numero ${salle_!!.numero}"
            + " pour le cours de ${cours_!!.matiere}")
    }
}